### PR TITLE
update CL doc

### PR DESCRIPTION
**Pull request by conglongli**
To reflect some changes on CL implementation/tuning strategy.

### CODE REVIEW

**Review by tjruwase — APPROVED** [review]

File: docs/_tutorials/curriculum-learning.md [hunk 1]
```diff
@@ -98,10 +102,35 @@ The `difficulty` is a list of difficulty levels to be used during schedule. The
...
+{: .notice--warning}
+
+We provide two curriculum learning examples for Megatron-LM GPT-2 pre-training:
+
+The first one is at [Megatron-DeepSpeed/tree/main/examples/curriculum_learning](https://github.com/microsoft/Megatron-DeepSpeed/tree/main/examples/curriculum_learning). This integration is based on a newer Megatron-LM fork, and only this curriculum learning example supports pipeline parallelism. However, currently (10/29/2021) we haven't verified ZeRO-2 and ZeRO-3 on this fork. Overall, we highly recommend you to use this example if your model does not require ZeRO-2/3.
+
+The second one is at [DeepSpeedExamples/Megatron-LM-v1.1.5-ZeRO3/curriculum_learning/](https://github.com/microsoft/DeepSpeedExamples/tree/master/Megatron-LM-v1.1.5-ZeRO3/curriculum_learning). This integration is based on a older Megatron-LM hard copy that we will eventually deprecate and this curriculum learning example does not support pipeline parallelism. We recommend you to ONLY use this example if your model requires ZeRO-2/3.
```
Comment (minjiaz) [inline docs/_tutorials/curriculum-learning.md:113]: "a older" -> "an older".

**Review by minjiaz — APPROVED** [review]
Conglong, the changes overall look great! I added minor comments.

File: docs/_tutorials/curriculum-learning.md [hunk 2]
```diff
@@ -2,7 +2,11 @@
...
+{: .notice--info}
+
+In this tutorial, we introduce DeepSpeed's curriculum learning-based data pipeline, which presents easier or simpler examples earlier during training. By enabling stable training with 8x/4x larger batch size/learning rate (whereas the baseline approach struggles with training divergence), we observe that curriculum learning (based on sequence length) provides stable and 3.3x faster GPT-2 pre-training (tested on 117M and 1.5B parameters), together with better token-wise convergence speed and zero-shot WikiText-103/LAMBADA evaluation results. In addition, since curriculum learning only affect the data pipeline, its benefit is complementary to many DeepSpeed features and other system optimization techniques. For example, curriculum learning is compatible with DeepSpeed's [ZeRO Redundancy Optimizer](/tutorials/zero/), [ZeRO-Offload](/tutorials/zero-offload/), and [3D Parallelism](/tutorials/pipeline/).
```
[the remark this answers is not in the document]
Comment (conglongli) [reply docs/_tutorials/curriculum-learning.md:9]: It is 3.3x for 117M and 2.9x for 1.5B, not that different so I think we can just say 3.3x here :)

File: docs/_tutorials/curriculum-learning.md (hunk 1 above)
Comment (conglongli) [reply docs/_tutorials/curriculum-learning.md]: Thanks! Solved.

**Pull request merged**